apps/tenant-management-webapp-e2e/src/integration/events/events.step.ts

```diff
@@ -175,3 +175,257 @@ Then('the user views the details of {string} under Core streams', function (stre
+  } else {
+    expect(viewOrNot).to.be.oneOf(['views', 'should not view']);
+  }
+});
+
+//Find stream  with name, Subscriber role(s)
+//Input: stream name, role(s) in a string separated with comma
+//Return: row number if the stream is found; zero if the stream isn't found
+function findStream(streamName, role) {
+  return new Cypress.Promise((resolve, reject) => {
+    try {
+      let rowNumber = 0;
+      const subscriberRoles = role.split(',');
+
+      const targetedNumber = subscriberRoles.length + 1; // Name, roles need to match to find the stream
+      eventsObj
+        .streamTableBody()
+        .find('tr')
+        .then((rows) => {
+          rows.toArray().forEach((rowElement) => {
+            let counter = 0;
+            if (rowElement.cells[0].innerHTML.includes(streamName)) {
+              counter = counter + 1;
+            }
+            subscriberRoles.forEach((sRole) => {
+              if (rowElement.cells[1].innerHTML.includes(sRole.trim())) {
+                counter = counter + 1;
+              }
+            });
+            Cypress.log({
+              name: 'Number of matched items for row# ' + rowElement.rowIndex + ': ',
+              message: String(String(counter)),
+            });
+            if (counter == targetedNumber) {
+              rowNumber = rowElement.rowIndex;
+            }
+          });
+          Cypress.log({
+            name: 'Row number for the found file type: ',
+            message: String(rowNumber),
+          });
+          resolve(rowNumber);
+        });
+    } catch (error) {
+      reject(error);
+    }
+  });
+}
+
+Then('the user {string} the stream of {string}, {string}', function (viewOrNot, streamName, role) {
+  findStream(streamName, role).then((rowNumber) => {
+    switch (viewOrNot) {
+      case 'views':
+        expect(rowNumber).to.be.greaterThan(0, 'Stream of ' + streamName + ', ' + role + ' has row #' + rowNumber);
+        break;
+      case 'should not view':
+        expect(rowNumber).to.equal(0, 'Stream of ' + streamName + ', ' + role + ' has row #' + rowNumber);
+        break;
+      default:
+        expect(viewOrNot).to.be.oneOf(['views', 'should not view']);
+    }
+  });
+});
+
+Then(
+  'the user views the stream details of {string}, {string}, {string}, {string}',
+  function (streamName, description, event, role) {
+    eventsObj.streamDetails(streamName).should('contain', streamName);
+    eventsObj.streamDetails(streamName).should('contain', description);
+    if (role == 'public') eventsObj.streamDetails(streamName).should('contain', '"publicSubscribe": true');
+    else {
+      eventsObj.streamDetails(streamName).should('contain', '"publicSubscribe": false');
+    }
+    const roles = role.split(',');
+    for (let i = 0; i < roles.length; i++) {
+      eventsObj
+        .streamDetails(streamName)
+        .invoke('text')
+        .then((roleDetails) => {
+          const subscriberRoles = roles[i].split(',')[0].trim();
```

Comment:
Why do we need another .split(',') here (split() has been done at line 315)? I believe roles[i].trim() should be enough as there should be no "," in the variable  already.

Reply:
yes it needed it there